cmd/containerd/command/service_windows.go

```diff
@@ -379,3 +384,54 @@ func removePanicFile() {
 		}
 	}
 }
+
+func logFileInChunks(path string) error {
+	fileReader, err := os.Open(path)
+	if err != nil {
+		return err
+	}
+	defer fileReader.Close()
+
+	st, err := fileReader.Stat()
+	if err != nil {
+		return err
+	}
+
+	if st.Size() == 0 {
+		return nil
+	}
+
+	// There seems to be a limit for the message length, which is somewhere between 16-32KB.
+	// Set the chunkSize to 16KB, which is the max power of 2 that seems to work.
+	chunkSize := 16 * 1024
+	numChunks := st.Size() / int64(chunkSize)
+	if st.Size()%int64(chunkSize) > 0 {
+		numChunks++
+	}
+	buf := make([]byte, chunkSize)
+
+	// Set reader size (buffer size) to 1MB to issue less reads.
+	readerSize := 1024 * 1024
+	reader := bufio.NewReaderSize(fileReader, readerSize)
+	logrus.WithFields(logrus.Fields{
+		"filename":   path,
+		"chunkCount": numChunks,
+	}).Debug("Begin logging file chunks")
+
+	indx := 0
+	for {
+		n, err := reader.Read(buf)
+		if err != nil {
+			if err != io.EOF {
+				logrus.WithError(err).Error("failed to read file chunks")
+			}
+			break
+		}
+		logrus.WithFields(logrus.Fields{
+			"index": indx,
```

Comment:
We don't ever increment `indx` so it'll always log every chunk with index 0.

Reply:
sharp eyes 😄 thanks!